File: random_walk/random_walk.py
import math
import random

def random_walk(n=10000, l=1):
    x = [0]
    y = [0]
    for i in range(1, n+1):
        theta = 2 * math.pi * random.uniform(0, 1)
        dx = l * math.cos(theta)
        dy = l * math.sin(theta)
        x.append(x[i-1] + dx)
        y.append(y[i-1] + dy)
    return x, y

File: random_walk/random_walk.js
function randomWalk(n = 10000, l = 1) {
    const x = [0];
    const y = [0];

    for (let i = 1; i <= n; i++) {
        const theta = 2 * Math.PI * Math.random();
        const dx = l * Math.cos(theta);
        const dy = l * Math.sin(theta);
        x.push(x[i - 1] + dx);
        y.push(y[i - 1] + dy);
    }

    return { x, y };
}